src/Microsoft.AspNet.OData.Shared/Builder/StructuralTypeConfiguration.cs

```diff
@@ -58,7 +57,9 @@ protected StructuralTypeConfiguration(ODataModelBuilder modelBuilder, Type clrTy
             ClrType = clrType;
             ModelBuilder = modelBuilder;
             _name = clrType.EdmName();
-            _namespace = clrType.Namespace ?? DefaultNamespace;
+
+            // Use the namespace is uder specified one, otherwise use CLR Namespace. If CLR Namespace is null we fallback to "Default"
+            _namespace = modelBuilder.HasAssignedNamespace ? modelBuilder.Namespace : clrType.Namespace ?? modelBuilder.Namespace;
```

Comment:
It seems it changes the previous logic for the namespace. It will change all the structural type namespace. 

By the way, do you want to change the EnumType namespace?

Reply:
I thought Enums were also going through Structural namespace I will make the appropriate change there as well.

Yes the intent indeed is to change the logic. As explained in the issue, today's approach messes with the auto builder and there is no simple way to put everything into same namespace. Builder's namespace has no value unless you are adding functions or actions. This PR is aiming to change that.